### PR TITLE
Submit spot requests by region

### PR DESCRIPTION
**Pull request by jhford**
Let's do each region on its own.

### CODE REVIEW

**Comment by jhford** [comment]
Note that this has a lot of #98 in it, it's meant to be landed on top of that.

File: src/provision.js [hunk 1]
```diff
@@ -285,42 +285,51 @@ class Provisioner {
...
-    forSpawning = forSpawning.slice(0, 400);
+    let byRegion = {};
+    for (let x of forSpawning) {
+      assert(x.bid && x.bid.region);
+      let r = x.bid.region;
+      if (!byRegion[r]) {
+        byRegion[r] = [x];
+      } else {
+        byRegion[r].push(x);
+      }
```
Comment (jonasfj) [inline src/provision.js]: You don't need this much code... or a mapping from region to requests...
You just need:
```js
let regions = _.uniq(_.map(forSpawning, ['bid', 'region']));
regions.map(region => _.filter(forSpawning, [['bid', 'region'], region]));

await Promise.all(regions.map(async (forSpawning) => {
  forSpawning = forSpawning.slice(0, 400);
  for (let toSpawn of inRegion) {
  ....
}));
```

**Comment by jonasfj** [comment]
# `r+`

**Comment by jhford** [comment]
f8b202b34f4e3709f519a478d97fc6eb8990b281 deployed this

**Pull request closed without merging**